### PR TITLE
Energy shield sounds

### PR DESCRIPTION
**Pull request by Generalcamo**
This PR replaces the energy shield's reusage of the energy blade, with new sounds ported from bay (See https://github.com/Baystation12/Baystation12/pull/32715)

This also optimizes the shield blocking code, and includes a few new macros to that effect. This PR originally intended to also optimize the shield reflection code, however that has been pushed back to another PR I may make in the future.

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

| Path | Original Author | License |
| sound/items/shield/energy/shield-loop.ogg | Spookerton (Baystation12) | GNU Affero General Public License v3 |
| sound/items/shield/energy/shield-start.ogg | Spookerton (Baystation12) | GNU Affero General Public License v3 |
| sound/items/shield/energy/shield-stop.ogg | Spookerton (Baystation12) | GNU Affero General Public License v3 |

### CODE REVIEW

**Comment by Generalcamo** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -144,105 +139,114 @@
...
+/obj/item/shield/energy/update_icon()
+	icon_state = "eshield[active]"
+	if(active)
+		set_light(1.5, 1.5, "#006AFF")
+	else
+		set_light(0)
+		
+/obj/item/shield/energy/attack_self(mob/living/user)
+	var/time = world.time
+	if (time < next_action)
```
Comment (SleepyGemmy) [inline code/game/objects/items/weapons/shields.dm:164]:
```suggestion
	if(time < next_action)
```

File: code/game/objects/items/weapons/shields.dm [hunk 2]
```diff
@@ -144,105 +139,114 @@
...
+		return
+	next_action = time + 3 SECONDS
+	active = !active
+	if (active)
```
Comment (SleepyGemmy) [inline code/game/objects/items/weapons/shields.dm:168]:
```suggestion
	if(active)
```

File: code/game/objects/items/weapons/shields.dm [hunk 3]
```diff
@@ -144,105 +139,114 @@
...
-	if(!active)
-		return FALSE //turn it on first!
+/obj/item/shield/energy/handle_shield(mob/user, on_back, damage, atom/damage_source = null, mob/attacker = null, def_zone = null, attack_text = "the attack")
+	var/shield_dir = on_back ? user.dir : reverse_dir[user.dir]
 
-	if(user.incapacitated())
+	if(!active || user.incapacitated() || !(check_shield_arc(user, shield_dir, damage_source, attacker)))
 		return FALSE
-
+
```
Comment (SleepyGemmy) [inline code/game/objects/items/weapons/shields.dm:182]:
```suggestion

```
i.e. newline with no whitespace.

File: html/changelogs/Generalcamo - EnergyShields.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#   experiment
+#   balance
+#   admin
+#   backend
+#   security
+#   refactor
+#################################
+
+# Your name.
+author: GeneralCamo, Spookerton
```
Comment (SleepyGemmy) [inline html/changelogs/Generalcamo - EnergyShields.yml:30]:
```suggestion
author: GeneralCamo
```
don't need to add them as an author unless you want to.

Comment (Generalcamo) [reply html/changelogs/Generalcamo - EnergyShields.yml:30]: Being that they authored the sounds, it's best we do so.

**Review by SleepyGemmy — APPROVED** [review]

**Comment by Spookerton** [comment]
Just noting that the sounds here would be under CC3-by-sa, not AGPL, because of general repo licensing. As the author, please consider them PD.

**Pull request merged**